### DISCUSSION TITLE
Lucene query runtime

### DISCUSSION THREAD

**Discussion by zhiqihuang** [castorini/pyserini#1938] [Q&A] [answered]
Hi,

When running Lucene, like the following:

```
python -m pyserini.search.lucene \
  --index indexes/sample_collection_jsonl \
  --topics tests/resources/sample_queries.tsv \
  --output run.sample.txt \
  --bm25
```

Is there a way to get the average runtime of each query? Thank you!

**Comment by lintool** [comment, marked as answer]
Hi @zhiqihuang - sorry for the late reply, just getting to this now.

Unfortunately, there isn't anything built-in to measure query latency. You'll have to write some custom code, but this shouldn't be too difficult to do.

**Reply by zhiqihuang** [reply]
No worries. I write custom code to measure runtime. Before that, I wondered if there was such a built-in feature. Thank you.
